src/content_cao.cpp

```diff
@@ -716,12 +716,26 @@ void GenericCAO::updateLightNoCheck(u8 light_at_pos)
 	}
 }
 
-v3s16 GenericCAO::getLightPosition()
+m4x4f GenericCAO::getTransform()
 {
-	if (m_is_player)
-		return floatToInt(m_position + v3f(0, 0.5 * BS, 0), BS);
+	scene::ISceneNode *node = getSceneNode();
+	if (node)
+		return m4x4f(node->getAbsoluteTransformation());
+	
+	return m4x4f( m4x4f::EM4CONST_IDENTITY );
```

Comment:
`m4x4f(m4x4f::EM4CONST_IDENTITY);` (if the initialization muse be kept)
https://dev.minetest.net/Code_style_guidelines